stumpy/core.py

```diff
@@ -2885,3 +2885,78 @@ def max_distance(D):
         candidate_idx = np.argmin(D)
 
     return np.array(matches, dtype=object)
+
+
+@njit(
+    # "(i8, i8, i8, i8[:], i8)"
+    fastmath=True,
+    parallel=True,
+)
+def _get_tiles(m, n_A, n_B, diags, tile_length):
+    """
```

Comment:
@alvii147 
I was wondering if you would like to remove `tile_length` from the arguments. (As you said, it is config. variable now. right?)

@seanlaw @alvii147 
Does this function make sense for any arbitrary `diag` range? I mean...the `diag` here should be calculated like this:
```
if ignore_trivial:
        diags = np.arange(excl_zone + 1, n_A - m + 1, dtype=np.int64)
    else:
        diags = np.arange(-(n_A - m + 1) + 1, n_B - m + 1, dtype=np.int64)
```
right?

And, we cannot pass a partial range of diags.  So, isn't it better to calculate `diags` inside this function and return both? Or, if we want to avoid mixing things and want to make sure this function just do one thing, we may just repeat the calculation of `diags` inside this function. We may repeat ourselves here but we reduce the arguments of `_get_tiles`.

So, do you think we should change the arguments to something like: `_get_tiles(m, n_A, n_B=None, ignore=Trivial)`?

Any thoughts?

Reply:
I think by instinct I would want to keep these things separate. I feel like `_get_tiles` should only get tiles information, and not have any logic of deciding what diagonals are to be traversed, or what tile length to use. I like keeping it more general purpose.

@seanlaw curious to hear what you think here.

Reply:
> I feel like `_get_tiles` should only get tiles information, and not have any logic of deciding what diagonals are to be traversed, or what tile length to use. I like keeping it more general purpose.

Right... I also checked out the its test function and I realized it can accept any range of `diags` and return their corresponding `tiles`. Your justification sounds reasonable.

Reply:
> curious to hear what you think here. 

I like that it is generic and, more importantly, does not depend on any internal state like a `config` variable. 

Unrelated: I've been wondering if the name `tile_size` (or something else) is a better term than `tile_length`?

Reply:
> Unrelated: I've been wondering if the name tile_size (or something else) is a better term than tile_length?

Agreed, this makes sense, done! I did leave the word `length` in the docstrings though, feels like `maximum length of a tile` makes it clear that we're talking about a size of a 2D object.